lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EnsembleCalibration.py

```diff
@@ -47,8 +47,21 @@
 from improver.utilities.warnings_handler import ManageWarnings
 
 IGNORED_MESSAGES = ["Collapsing a non-contiguous coordinate.",
-                    "Not importing directory .*sphinxcontrib'"]
-WARNING_TYPES = [UserWarning, ImportWarning]
+                    "Not importing directory .*sphinxcontrib'",
+                    "The pandas.core.datetools module is deprecated",
+                    "numpy.dtype size changed",
+                    "The statsmodels can not be imported",
+                    "invalid escape sequence",
+                    "can't resolve package from",
+                    "Collapsing a non-contiguous coordinate.",
+                    "Minimisation did not result in"
+                    " convergence",
+                    "\nThe final iteration resulted in a percentage "
+                    "change that is greater than the"
+                    " accepted threshold "]
+WARNING_TYPES = [UserWarning, ImportWarning, FutureWarning, RuntimeWarning,
```

Comment:
The reason that there are multiple UserWarning and ImportWarning is because these need to match up with the messages in the IGNORED_MESSAGES list i.e.:

IGNORED_MESSAGES | WARNING_TYPES
------------------------|-------------------
"Collapsing a non-contiguous coordinate." | UserWarning
"Not importing directory .*sphinxcontrib'" | ImportWarning
"The pandas.core.datetools module is deprecated" | FutureWarning

Therefore there needs to be the same number of warnings as there are IGNORED_MESSAGES, so that's why it looks like the warnings are repeated, as the different messages are associated with different types of warning. The default warning is a UserWarning, so in some instances, no WARNING_TYPES list is provided, if all of the messages are UserWarnings. 

You're right that it's not obvious that these two lists need to be the same length. Maybe a dictionary would be an alternative interface e.g.

MESSAGES_AND_WARNINGS = {
    "Collapsing a non-contiguous coordinate.": UserWarning,
    "Not importing directory .*sphinxcontrib'": ImportWarning,
    ...
}
or a list of tuples:
MESSAGES_AND_WARNINGS = [(UserWarning, "Collapsing a non-contiguous coordinate."), (ImportWarning, "Not importing directory .*sphinxcontrib'"), ...]

Reply:
No I don't think it is obvious, so it might be useful if it was easier to understand that is the reason there are multiple similar warnings?